python/dgl/distributed/dist_graph.py

```diff
@@ -600,7 +596,7 @@ def ndata(self):
             The data view in the distributed graph storage.
         """
         assert len(self.ntypes) == 1, "ndata only works for a graph with one node type."
-        return self._ndata
+        return NodeDataView(self)
```

Comment:
It looks like creating `NodeDataView` or `EdgeDataView` will also create `DistTensor`

https://github.com/dmlc/dgl/blob/949f87cc1df76980ce2af2ac871f7f518463b2fc/python/dgl/distributed/dist_graph.py#L190-L196

How costly are they? This may be an issue because `DistGraph` is typically created only once but `ndata` and `edata` may be invoked many times.

Reply:
`DistTensor` are referenced by `g._ndata_store`, so will not be created/destroyed in each call.

Reply:
I refactored the data store init logic. this is a bit radical. pls take a look

Reply:
`self._ndata/_edata` is not defined yet because I think it's trivial to instantiate `NodeDataView` when calling `g.ndata`. And `g.nodes['_N']` and `g.ndata` have the same behavior: always instantiate a new one.
In original code, a new `NodeDataView` is always instantiated when calling `g.nodes['_N']`.

Reply:
always instantiating a new `NodeDataView` and creating `DistTensor` by iterating over existing names is more conservative than what I does now.